#! /bin/sh /usr/share/dpatch/dpatch-run
## 104_bitmap_type_print.dpatch by Bryce Harrington <bryce@ubuntu.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Bitmap-type print to printer and preview produces low resolution output
## DP: https://bugs.edge.launchpad.net/ubuntu/+source/inkscape/+bug/258916

@DPATCH@

diff -Nurp patched/src/ui/dialog/print.cpp working/src/ui/dialog/print.cpp
--- patched/src/ui/dialog/print.cpp	2009-04-07 22:55:14.000000000 -0700
+++ working/src/ui/dialog/print.cpp	2009-04-07 23:33:02.000000000 -0700
@@ -58,8 +58,8 @@ draw_page (GtkPrintOperation *operation,
 
             sp_export_png_file(junk->_doc, tmp_png.c_str(), 0.0, 0.0,
                 width, height,
-                (unsigned long)(width * PT_PER_IN / PX_PER_IN),
-                (unsigned long)(height * PT_PER_IN / PX_PER_IN),
+                (unsigned long)(width * dpi / PX_PER_IN),
+                (unsigned long)(height * dpi / PX_PER_IN),
                 dpi, dpi, bgcolor, NULL, NULL, true, NULL);
 
             // This doesn't seem to work:
@@ -74,10 +74,14 @@ draw_page (GtkPrintOperation *operation,
             {
                 Cairo::RefPtr<Cairo::ImageSurface> png = Cairo::ImageSurface::create_from_png (tmp_png);
                 cairo_t *cr = gtk_print_context_get_cairo_context (context);
+		cairo_matrix_t m;
+		cairo_get_matrix(cr, &m);
+		cairo_scale(cr, PT_PER_IN / dpi, PT_PER_IN / dpi);
                 // FIXME: why is the origin offset??
                 cairo_set_source_surface(cr, png->cobj(), -16.0, -16.0);
+		cairo_paint(cr);
+		cairo_set_matrix(cr, &m);
             }
-            cairo_paint(gtk_print_context_get_cairo_context (context));
 
             // Clean up
             unlink (tmp_png.c_str());
